### PR TITLE
feat: main net flag

### PR DESCRIPTION
**Pull request by anatollupacescu**
Allows for providing network alias instead of IDs.

This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2159)

### CODE REVIEW

File: cmd/bee/cmd/cmd.go [hunk 1]
```diff
@@ -243,6 +245,8 @@ func (c *command) setAllFlags(cmd *cobra.Command) {
 	cmd.Flags().Uint64(optionNameBlockTime, 15, "chain block time")
 	cmd.Flags().String(optionNameSwapDeploymentGasPrice, "", "gas price in wei to use for deployment and funding")
 	cmd.Flags().Duration(optionWarmUpTime, time.Minute*10, "time to warmup the node before pull/push protocols can be kicked off.")
+	cmd.Flags().Bool(optionNameMainNet, false, "triggers connect to main net bootnodes.")
```
Comment (Eknir) [inline cmd/bee/cmd/cmd.go:248]: Default is true for mainnet

**Comment by Eknir** [comment]
Looks good! Missing the default connection to our bootnode cluster:

When mainnet (networkID 1), connect to /dnsaddr/mainnet.ethswarm.org 
When mainnet (networkID 10), connect to /dnsaddr/mainnet.ethswarm.org 

Also, please ensure that we update the commented-out config for our packages, update the documentation, validate configuration in docker packages.

**Comment by ldeffenb** [comment]
> When mainnet (networkID 1), connect to /dnsaddr/mainnet.ethswarm.org
> When mainnet (networkID 10), connect to /dnsaddr/mainnet.ethswarm.org

Huh?  Shouldn't the second be "When goerli"?  Shouldn't these be different dnsnames?

**Comment by Eknir** [comment]
@significance , heads up for configuration change. Please update the docs.

**Review by Eknir — APPROVED** [review]
Approving the logic, which I reviewed together with Anatol. It would be good if another reviewer validates code implementation.

I am glad to do a end-user review on the workings tomorrow.

**Comment by acud** [comment]
> > When mainnet (networkID 1), connect to /dnsaddr/mainnet.ethswarm.org
> > When mainnet (networkID 10), connect to /dnsaddr/mainnet.ethswarm.org
> 
> Huh? Shouldn't the second be "When goerli"? Shouldn't these be different dnsnames?

:+1:

File: cmd/bee/cmd/start.go [hunk 2]
```diff
@@ -404,3 +419,33 @@ func (c *command) configureSigner(cmd *cobra.Command, logger logging.Logger) (co
...
+func parseBootnodes(main, test bool, bootnodes []string) ([]string, error) {
+	switch {
+	case len(bootnodes) > 0: // use the user provided values
+		return bootnodes, nil
+	case main:
+		return []string{"/dnsaddr/mainnet.ethswarm.org"}, nil
+	case test:
+		return []string{"/dnsaddr/testnet.ethswarm.org"}, nil
+	default:
+		return nil, errors.New("bootnode required when not connecting to the official main or test networks")
```
Comment (acud) [inline cmd/bee/cmd/start.go:449]: why?

Comment (esadakar) [reply cmd/bee/cmd/start.go:449]: standalone nodes do not require bootnodes, I believe

**Review by esadakar — APPROVED** [review]

File: cmd/bee/cmd/start.go [hunk 3]
```diff
@@ -404,3 +419,33 @@ func (c *command) configureSigner(cmd *cobra.Command, logger logging.Logger) (co
...
 	}, nil
 }
+
+func parseNetworks(main, test bool, networkID uint64) (uint64, error) {
+	switch {
+	case main && test:
+		return 0, errors.New("please provide either mainnet or testnet option")
+	case main && networkID != 1:
+		return 0, errors.New("provided network ID does not match mainnet")
+	case test && networkID != 10:
```
Comment (aloknerurkar) [inline cmd/bee/cmd/start.go:429]: These are supposed to be bitwise ands?

File: cmd/bee/cmd/cmd.go [hunk 4]
```diff
@@ -243,6 +245,8 @@ func (c *command) setAllFlags(cmd *cobra.Command) {
 	cmd.Flags().Uint64(optionNameBlockTime, 15, "chain block time")
 	cmd.Flags().String(optionNameSwapDeploymentGasPrice, "", "gas price in wei to use for deployment and funding")
 	cmd.Flags().Duration(optionWarmUpTime, time.Minute*10, "time to warmup the node before pull/push protocols can be kicked off.")
+	cmd.Flags().Bool(optionNameMainNet, true, "triggers connect to main net bootnodes.")
```
Comment (janos) [inline cmd/bee/cmd/cmd.go:248]: The main net is set to true, but default network id is 10 on line 217, which should result with an error from parseNetworks, by default settings.

**Review by janos — COMMENTED** [review]
Bootnodes selection should be handled in regards to the standalone flag, as already commented.

**Review by acud — APPROVED** [review]

**Pull request merged**